learning-bitmark/working-with-bitmarks/payment-for-bitmark-cli.md

```diff
@@ -0,0 +1,261 @@
+# Bitmark transaction payment
+
+In the Bitmark Property System, for each asset, each Bitmark Account is able to issue a bitmark for free, and has to pay for all the other transactions such as bitmark transfers or other bitmark issuances.
+
+The Bitmark CLI submits transactions to its connected node via RPC. Once receiving these transactions, the node processes them and returned the corresponding payment ids along with the payment amounts and addresses. Only when the payment - either by bitcoin or litecoin - is executed, a transaction is updated from `Pending` to `Verified` and is ready to be mined. The payment transaction needs to send:
+
+* A correct amount
+
+* To the correct address
+
+* With the correct payment id added as bitcoin/litecoin OP_RETURN opcode.
+
+Because common wallets do not support attaching the Bitmark blockchain payment identifier to the ltc/btc transactions, we provide an app called the Bitmark Wallet to pay for the Bitmark transactions come from the Bitmark CLI.
+
+<br>
+<br>
+
+## Bitmark Wallet
+
+The Bitmark Wallet is a dual currency wallet, supporting both bitcoin and litecoin. It allows sending multiple payments to different addresses and attaching a hex data item to a transaction using the OP_RETURN feature of bitcoin/litecoin.
+
+The wallet is written in Go and cannot work standalone, it can only send the bitcoin and litecoin transactions via a correct configured bitcoind and litecoind. Depending on the configuration of bitcoind and litecoind, the payment transactions are submitted to the mainnet coins or testnet coins.
+
```

Comment:
Probably need some pointers to tutorials for installing bitcoind and/or litecoind, if it hasn't already been explained in the tutorial.

Reply:
To install the bitmark wallet, the following things need to be installed:
* Git has been installed.

* Go has been installed.

* Python3 is installed.

* bitcoind/litecoind have been installed.
Why do we just point to bitcoind / litecoind tutorials but not the other?  In addition, the bitmark-cli is to test the bitmarkd so its users intend to be advance users. We would like to try with the assumption that they understand about the blockchain and can handle these prerequisites first. If there is any question on this section we could improve later.

Reply:
My feeling is that Git, Go, and Python3 are all pretty easy installs, mostly just a package add, while bitcoind/litecoind is much more in-depth. So, that's why I suggested the one, not the other. Even if the users are advanced, it doesn't hurt to give them a link to the reference (and I was just suggesting a link to the public instructions on installing those, not writing it ourselves).